packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1438 @@
+      _buffer
+        ..write('\n')
+        ..write(prefixOtherLines)
+        ..write(parts[i]);
+    }
+
+    if (lineTerminated)
+      _buffer.write('\n');
+
+    _atLineStart = lineTerminated;
+  }
+
+  /// Write text assuming the text already obeys the specified prefixes for the
+  /// first and subsequent lines.
+  void writeRaw(String text) {
+    if (text.isEmpty)
+      return;
+    _buffer.write(text);
+    _atLineStart = text.endsWith('\n');
+  }
+
+
+  /// Write a line assuming the line obeys the specified prefixes. Ensures that
+  /// a newline is added if one is not present.
+  /// The same as [writeRaw] except a newline is added at the end of [line] if
+  /// one is not already present.
+  ///
+  /// A new line is not added if the input string already contains a newline.
+  void writeRawLine(String line) {
+    if (line.isEmpty)
+      return;
+    _buffer.write(line);
+    if (!line.endsWith('\n'))
+      _buffer.write('\n');
+    _atLineStart = true;
+  }
+
+  @override
+  String toString() => _buffer.toString();
+}
+
+/// Defines diagnostics data for an [Object].
+///
+/// DiagnosticsNode provides a high quality multi-line string dump via
+/// [toStringDeep]. The core members are the [name], [header], [getProperties],
+/// [object], and [getChildren]. All other members exist typically to provide
+/// hints for how [toStringDeep] and debugging tools should format output.
+abstract class DiagnosticsNode {
+  DiagnosticsNode({
+    @required this.name,
+    @required this.header,
+    this.style: DiagnosticsTreeStyle.normal,
+    bool hidden: false,
+    this.showNull: false,
+    this.showName: true,
+    this.showSeparator: true,
+    this.emptyDescription,
+    this.defaultValue,
+  })
+      : _hidden = hidden {
+    // A name ending with ':' indicates that the user forgot that the ':' will
+    // be automatically added for them when generating descriptions of the
+    // property.
+    assert(name == null || !name.endsWith(':'));
+  }
+
+  /// Property with a [String] value.
+  ///
+  /// If `value` is not a string but you wish to specify a custom [String]
+  /// describing the `value` use [DiagnosticsNode.objectProperty] and specify a
+  /// custom header instead of using an instead of using an artificial [String]
+  /// value. That allows debugging to access the raw value of the property being
+  /// inspected as well as the [String] description.
+  /// If `showNull` is true, the property is shown even if its `value` is null.
+  /// `header` overrides the normal description of the property value.
+  /// `showNull` indicates whether the property should still be shown if its
+  /// `value` is null. `showName indicates whether the property name should be
+  /// shown. `hidden` specifies whether the property should be hidden or shown.
+  /// `nullDescription` provides a description of the property value if it is
+  /// null. If `quoted` is true, the `value` is enclosed
+  /// in double quotes.
```

Comment:
This will all turn into one big paragraph; if you meant each value to be on its own line you have to separate the paragraphs with a blank line. Maybe put each argument on a bulleted line?